src/cmd/update.rs

```diff
@@ -0,0 +1,68 @@
+use std::fs;
+use std::fs::File;
+use std::fs::Permissions;
+use std::os::unix::prelude::*;
+
+use anyhow::Result;
+use serde::Deserialize;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const RELEASES_LIST: &str = "https://api.github.com/repos/sbstp/kubie/contents/releases/linux?ref=master";
+const RELEASE_BASE_URL: &str = "https://github.com/sbstp/kubie/raw/master/releases/linux/amd64";
+const FILENAME: &str = "kubie";
+
+#[derive(Debug, Deserialize)]
+struct TreeUrl {
+    git_url: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Tree {
+    tree: Vec<KubieVersion>,
+}
+
+#[derive(Debug, Deserialize)]
+struct KubieVersion {
+    path: String,
+}
+
+pub fn update() -> Result<()> {
+    let latest_version = get_latest_version()?;
+    if latest_version.eq(&format!("v{}", VERSION)) {
+        println!("Kubie is up-to-date : v{}", VERSION);
+    } else {
+        println!(
+            "A new version of Kubie is available ({}), the new version will be automatically installed...",
+            latest_version
+        );
+        let resp = attohttpc::get(format!("{}/{}/{}", RELEASE_BASE_URL, latest_version, FILENAME)).send()?;
+        if resp.is_success() {
+            let file = File::create(FILENAME)?;
+            let _n = resp.write_to(file)?;
+            let old_file = which::which(FILENAME).unwrap();
```

Comment:
Instead of doing an unwrap here, can you bubble up the error? Normally it should always work, especially since kubie itself it calling this, but it can fail for other reasons.

You can use the `.context` method provided by `anyhow` to add context to the error, like done [here](https://github.com/sbstp/kubie/blob/a0cd6faf93aa9501aa0ab543f3a4f42493468476/src/session.rs#L42) and then bubble up the error with `?`.

Reply:
I didn't used the ".context" on this, but I used the `env::current_exe().expect("could not get own binary path");` instead.